File: flutter_demo/ios/Runner/APNParamsModel.m
//
//  APNParamsModel.m
//  Runner
//
//  Created by NingSpeals on 2022/6/30.
//

#import "APNParamsModel.h"

@implementation APNParamsModel
- (instancetype)initWithDict:(NSDictionary *)dict{
    if (self == [super init]) {
        self.apnType = [dict[@"apnType"] integerValue];
        self.apnLen = [dict[@"apnLen"] integerValue];
        self.apn = dict[@"apn"];
        self.userNameLen = [dict[@"userNameLen"] integerValue];
        self.userName = dict[@"userName"];
        self.pswLen = [dict[@"pswLen"] integerValue];
        self.psw = dict[@"psw"];
        self.imei = dict[@"imei"];
        self.imsi = dict[@"imsi"];
        self.csq = [dict[@"csq"] integerValue];
    }
    return self;
}

@end

File: flutter_demo/lib/Home/ChargerLink/APNParamsModel.dart
// To parse this JSON data, do
//
//     final apnParamsModel = apnParamsModelFromJson(jsonString);

import 'dart:convert';

APNParamsModel apnParamsModelFromJson(String str) =>
    APNParamsModel.fromJson(json.decode(str));

String apnParamsModelToJson(APNParamsModel data) => json.encode(data.toJson());

class APNParamsModel {
  APNParamsModel({
    required this.apn,
    required this.apnLen,
    required this.psw,
    required this.pswLen,
    required this.userName,
    required this.userNameLen,
    required this.apnType,
    required this.imei,
    required this.imsi,
    required this.csq,
  });

  int apnLen;
  int pswLen;
  dynamic userName;
  int apnType;
  int userNameLen;
  dynamic psw;
  String imei;
  String imsi;
  int csq;
  dynamic apn;

  factory APNParamsModel.fromJson(Map<String, dynamic> json) => APNParamsModel(
        apnLen: json["apnLen"],
        pswLen: json["pswLen"],
        userName: json["userName"],
        apnType: json["apnType"],
        userNameLen: json["userNameLen"],
        psw: json["psw"],
        imei: json["imei"],
        imsi: json["imsi"],
        csq: json["csq"],
        apn: json["apn"],
      );

  Map<String, dynamic> toJson() => {
        "apnLen": apnLen,
        "pswLen": pswLen,
        "userName": userName,
        "apnType": apnType,
        "userNameLen": userNameLen,
        "psw": psw,
        "imei": imei,
        "imsi": imsi,
        "csq": csq,
        "apn": apn,
      };
}
